frigate/detectors/plugins/rknn.py

```diff
@@ -21,96 +15,146 @@ def hide_warnings(func):
 
 supported_socs = ["rk3562", "rk3566", "rk3568", "rk3588"]
 
+supported_models = ["^default-fp16-yolonas_[sml]$"]
+
+default_model = "default-fp16-yolonas_s"
+
+model_chache_dir = "/config/model_cache/rknn_cache/"
+
 
 class RknnDetectorConfig(BaseDetectorConfig):
     type: Literal[DETECTOR_KEY]
-    core_mask: int = Field(default=0, ge=0, le=7, title="Core mask for NPU.")
+    num_cores: int = Field(default=0, ge=0, le=3, title="Number of NPU cores to use.")
+    purge_model_cache: bool = Field(default=True)
```

Comment:
what is this for exactly? is it to remove old RKNN model files or something that applies to new users as well?

Reply:
This applies to new users as well. I have to convert the models to rknn format using the same toolkit version that I use for inference. So, if I update the toolkit, the user needs to download a new updated model. This function removes older models, but the user can choose to disable this.

Reply:
can we do something like making the name of the converted file have the version in the name so a config property is not needed and the desirable behavior is just there by default?

Reply:
Yes, this is already the case. The naming scheme is this:
```
Filenames follow this pattern:
origin-quant-basename-soc-tk_version-rev.rknn
origin: From where comes the model? default: upstream repo; rknn: modifications from airockchip
quant: i8 or fp16
basename: e.g. yolonas_s
soc: e.g. rk3588
tk_version: e.g. v2.0.0
rev: e.g. 1
Full name could be: default-fp16-yolonas_s-rk3588-v2.0.0-1.rknn
```

However, this means that over time frigate will download more and more models and use up space.
Frigate will just purge model in the subdir `/config/model_cache/rknn_cache/". The user is not allowed to place his own models there. I will add a warning to the docs.

Reply:
Alright. Should I remove this function or set it to `false` by default?

Reply:
I would remove it